inc/connector/namespace.php

```diff
@@ -268,8 +268,14 @@ function pre_unschedule_event( $pre, $timestamp, $hook, $args, $wp_error = false
 
 	$job = $jobs[0];
 
-	// Delete it.
-	$job->delete();
+	// Check if the job is running more then 3 hours from the timestamp.
+	$is_delete_running = ( time() - $job->start ) > 10800;
+	if ( $is_delete_running ) {
+		$job->delete( [ 'delete_running' => $is_delete_running ] );
```

Comment:
Rather than being this opinionated in the Cavalcade code itself, could we introduce a filter `cavalcade/job-max-runtime` which defaults to `false`, but lets a project opt in to this behavior when set?

Reply:
The issue when I try to fix this on another project was, if the job is running status and stalled, it cannot be deleted from `wp_unschedule_event` triggered from cron delete from cli.

Although I agree on the max time out is up to project to decide.